terraform/aws-accounts/cloud-platform-ephemeral-test/account/opensearch.tf

```diff
@@ -0,0 +1,274 @@
+        Effect = "Allow"
+        Resource = [
+          "arn:aws:es:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:domain/${local.live_modsec_audit_domain}/*",
+        ]
+      },
+    ]
+  })
+}
+
+data "aws_iam_policy_document" "live_modsec_audit" {
+  statement {
+    actions = [
+      "es:*",
+    ]
+
+    resources = [
+      "arn:aws:es:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:domain/${local.live_modsec_audit_domain}/*",
+    ]
+
+    principals {
+      type        = "AWS"
+      identifiers = ["*"]
+    }
+
+    condition {
+      test     = "IpAddress"
+      variable = "aws:SourceIp"
+      values   = keys(local.allowed_live_1_ips)
+    }
+  }
+}
+
+# resource "random_password" "master_password" {
+#   length      = 16
+#   min_lower   = 1
+#   min_upper   = 1
+#   min_numeric = 1
+#   min_special = 1
+# }
+
+resource "aws_kms_key" "live_modsec_audit" {
+  description = "Used for OpenSearch: cp-live-modsec-audit"
+  key_usage   = "ENCRYPT_DECRYPT"
+  # policy = TBD if one needed for console access
+  bypass_policy_lockout_safety_check = false
+  deletion_window_in_days            = 30
+  is_enabled                         = true
+  enable_key_rotation                = false
+  multi_region                       = false
+}
+
+resource "aws_opensearch_domain" "live_modsec_audit" {
+  domain_name    = "cp-live-modsec-audit"
+  engine_version = "OpenSearch_2.5"
+
+  advanced_options = {
+    "rest.action.multi.allow_explicit_index" = "true"
+    "indices.query.bool.max_clause_count"    = 3000
+    "override_main_response_version"         = "true"
+  }
+
+  advanced_security_options {
+    enabled                        = true
+    anonymous_auth_enabled         = false
+    internal_user_database_enabled = false
+    master_user_options {
+      master_user_arn = aws_iam_role.os_access_role.arn
+    }
+    # master_user_options {
+    #   master_user_name     = "admin"
+    #   master_user_password = random_password.master_password.result
+    # }
+  }
+
+  cluster_config {
+    instance_type            = "r6g.xlarge.search"
+    instance_count           = "3"
+    dedicated_master_enabled = true
+    dedicated_master_type    = "m6g.large.search"
+    dedicated_master_count   = "3"
+    zone_awareness_enabled   = true
+
+    zone_awareness_config {
+      availability_zone_count = 3
+    }
+
+    warm_count   = 3
+    warm_enabled = true
+    warm_type    = "ultrawarm1.medium.search"
+
+    cold_storage_options {
+      enabled = true
+    }
+  }
+
+  ebs_options {
+    ebs_enabled = "true"
+    volume_type = "gp3"
+    volume_size = "500"
+    iops        = "3000"
+  }
+
+  domain_endpoint_options {
+    enforce_https       = true
+    tls_security_policy = "Policy-Min-TLS-1-2-2019-07" # default to TLS 1.2
+  }
+
+  access_policies = data.aws_iam_policy_document.live_modsec_audit.json
+
+  snapshot_options {
+    automated_snapshot_start_hour = 23
```

Comment:
this is deprecated in `aws_opensearch` resources

Reply:
Removed the deprecated option